src/SourceGenerators/System.Xaml/System.Xaml/ParsedMarkupExtensionInfo.cs

```diff
@@ -63,47 +52,40 @@ public static ParsedMarkupExtensionInfo Parse(string raw, IXamlNamespaceResolver
 				throw Error("Invalid markup extension attribute. It should begin with '{{', but was {0}", raw);
 			}
 
-			var ret = new ParsedMarkupExtensionInfo();
-			int idx = raw.LastIndexOf('}');
+			if (raw.Length >= 2 && raw[1] == '}')
+			{
+				throw Error("Markup extension can not begin with an '{}' escape: '{0}'", raw);
+			}
 
-			if (idx < 0)
+			var ret = new ParsedMarkupExtensionInfo();
+			if (raw[raw.Length - 1] != '}')
 			{
+				// Any character after the final closing bracket is not accepted. Therefore, the last character should be '}'.
+				// Ideally, we should still ran the entire markup through the parser to get a more meaningful error.
 				throw Error("Expected '}}' in the markup extension attribute: '{0}'", raw);
 			}
-				
-			raw = raw.Substring(1, idx - 1);
-			idx = raw.IndexOf(' ');
-			string name = idx < 0 ? raw : raw.Substring(0, idx);
 
-			XamlTypeName xtn;
-			if (!XamlTypeName.TryParse(name, nsResolver, out xtn))
+			var nameSeparatorIndex = raw.IndexOf(' ');
+			var name = nameSeparatorIndex != -1 ? raw.Substring(1, nameSeparatorIndex - 1) : raw.Substring(1, raw.Length - 2);
+			if (!XamlTypeName.TryParse(name, nsResolver, out var xtn))
 			{
 				throw Error("Failed to parse type name '{0}'", name);
 			}
 
 			var xt = sctx.GetXamlType(xtn) ?? new XamlType(xtn.Namespace, xtn.Name, null, sctx);
 			ret.Type = xt;
 
-			if (idx < 0)
+			if (nameSeparatorIndex < 0)
 				return ret;
 
-			var valueWithoutBinding = raw.Substring(idx + 1, raw.Length - idx - 1);
-
-			var vpairs = BindingMembersRegex.Matches(valueWithoutBinding)
-				.Cast<Match>()
-				.Select(m => m.Value.Trim())
-				.ToList();
+			var valueWithoutBinding = raw.Substring(nameSeparatorIndex + 1, raw.Length - 1 - (nameSeparatorIndex + 1));
```

Comment:
Probably `nameSeparatorIndex + 1` can be named somehow and cached into a variable
```suggestion
			var valueWithoutBinding = raw[(nameSeparatorIndex + 1)..^(1 + (nameSeparatorIndex + 1))];
```